### ISSUE TITLE
Investigate failing CA Tests

### ISSUE THREAD

**Issue by cdietrich** [eclipse/xtext-xtend#709]
https://ci.eclipse.org/xtext/job/xtext-xtend/job/master/250/#showFailuresLink

@holgerschill is this related to your changes?
are the tests order dependent?

**Comment by holgerschill** [comment]
Will investigate- can‘t see a relation yet. Today I am ill so I‘ll do that hopefully tomorrow.

Von meinem iPhone gesendet

**Comment by cdietrich** [comment]
Undoing your change seems to resolve it. so it seems to be related.

**Comment by cdietrich** [comment]
i am not sure why it does not fail on typefox jenkins.

**Comment by cdietrich** [comment]
from locally started 
```
1724 [main] WARN  org.eclipse.xtext.ui.editor.contentassist.AbstractJavaBasedContentProposalProvider  - Error in polymorphic dispatcher : type may not be an array type
java.lang.IllegalArgumentException: type may not be an array type
	at org.eclipse.xtext.xbase.typesystem.references.ParameterizedTypeReference.<init>(ParameterizedTypeReference.java:58)
	at org.eclipse.xtext.xbase.typesystem.references.StandardTypeReferenceOwner.newParameterizedTypeReference(StandardTypeReferenceOwner.java:162)
	at org.eclipse.xtext.xbase.typesystem.references.LightweightTypeReference.isAssignableFrom(LightweightTypeReference.java:563)
	at org.eclipse.xtext.xbase.ui.contentassist.XbaseProposalProvider$1.apply(XbaseProposalProvider.java:761)
	at org.eclipse.xtext.xbase.ui.contentassist.XbaseProposalProvider$1.apply(XbaseProposalProvider.java:1)
	at com.google.common.collect.Iterators$6.computeNext(Iterators.java:617)
	at com.google.common.collect.AbstractIterator.tryToComputeNext(AbstractIterator.java:145)
	at com.google.common.collect.AbstractIterator.hasNext(AbstractIterator.java:140)
	at com.google.common.collect.Iterators$6.computeNext(Iterators.java:615)
	at com.google.common.collect.AbstractIterator.tryToComputeNext(AbstractIterator.java:145)
	at com.google.common.collect.AbstractIterator.hasNext(AbstractIterator.java:140)
	at com.google.common.collect.Iterators.addAll(Iterators.java:366)
	at com.google.common.collect.Lists.newArrayList(Lists.java:165)
	at com.google.common.collect.Lists.newArrayList(Lists.java:150)
	at org.eclipse.xtext.xbase.ui.contentassist.XbaseProposalProvider.getFavoriteStaticFeatures(XbaseProposalProvider.java:887)
	at org.eclipse.xtext.xbase.ui.contentassist.XbaseProposalProvider.completeXMemberFeatureCall_Feature(XbaseProposalProvider.java:752)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.eclipse.xtext.util.PolymorphicDispatcher.invoke(PolymorphicDispatcher.java:302)
	at org.eclipse.xtext.ui.editor.contentassist.AbstractJavaBasedContentProposalProvider.invokeMethod(AbstractJavaBasedContentProposalProvider.java:300)
	at org.eclipse.xtext.ui.editor.contentassist.AbstractJavaBasedContentProposalProvider.completeAssignment(AbstractJavaBasedContentProposalProvider.java:205)
	at org.eclipse.xtext.ui.editor.contentassist.AbstractContentProposalProvider$DefaultContentAssistProcessorSwitch.caseAssignment(AbstractContentProposalProvider.java:66)
	at org.eclipse.xtext.ui.editor.contentassist.AbstractContentProposalProvider$DefaultContentAssistProcessorSwitch.caseAssignment(AbstractContentProposalProvider.java:1)
	at org.eclipse.xtext.util.XtextSwitch.doSwitch(XtextSwitch.java:144)
	at org.eclipse.emf.ecore.util.Switch.doSwitch(Switch.java:53)
	at org.eclipse.emf.ecore.util.Switch.doSwitch(Switch.java:69)
	at org.eclipse.xtext.ui.editor.contentassist.AbstractContentProposalProvider$DefaultContentAssistProcessorSwitch.accept(AbstractContentProposalProvider.java:72)
	at org.eclipse.xtext.ui.editor.contentassist.AbstractContentProposalProvider.createProposals(AbstractContentProposalProvider.java:104)
	at org.eclipse.xtext.ui.editor.contentassist.AbstractJavaBasedContentProposalProvider.createProposals(AbstractJavaBasedContentProposalProvider.java:275)
	at org.eclipse.xtext.ui.editor.contentassist.CompletionProposalComputer.exec(CompletionProposalComputer.java:52)
	at org.eclipse.xtext.ui.editor.contentassist.CompletionProposalComputer.exec(CompletionProposalComputer.java:1)
	at org.eclipse.xtext.resource.OutdatedStateManager.exec(OutdatedStateManager.java:91)
	at org.eclipse.xtext.ui.editor.model.XtextDocument$XtextDocumentLocker.internalReadOnly(XtextDocument.java:527)
	at org.eclipse.xtext.ui.editor.model.XtextDocument$XtextDocumentLocker.priorityReadOnly(XtextDocument.java:491)
	at org.eclipse.xtext.ui.editor.model.XtextDocument.priorityReadOnly(XtextDocument.java:147)
	at org.eclipse.xtext.ui.editor.contentassist.RepeatedContentAssistProcessor.computeCompletionProposals(RepeatedContentAssistProcessor.java:111)
	at org.eclipse.xtext.ui.editor.contentassist.RepeatedContentAssistProcessor.computeCompletionProposals(RepeatedContentAssistProcessor.java:80)
	at org.eclipse.xtext.ui.testing.ContentAssistProcessorTestBuilder.computeCompletionProposals(ContentAssistProcessorTestBuilder.java:510)
	at org.eclipse.xtext.ui.testing.ContentAssistProcessorTestBuilder.computeCompletionProposals(ContentAssistProcessorTestBuilder.java:501)
	at org.eclipse.xtext.ui.testing.ContentAssistProcessorTestBuilder.computeCompletionProposals(ContentAssistProcessorTestBuilder.java:491)
	at org.eclipse.xtext.ui.testing.ContentAssistProcessorTestBuilder.computeCompletionProposals(ContentAssistProcessorTestBuilder.java:479)
	at org.eclipse.xtext.ui.testing.ContentAssistProcessorTestBuilder.getProposalsAtCursorIndicator(ContentAssistProcessorTestBuilder.java:280)
	at org.eclipse.xtext.ui.testing.ContentAssistProcessorTestBuilder.assertProposalDisplayedAtCursor(ContentAssistProcessorTestBuilder.java:294)
	at org.eclipse.xtend.ide.tests.contentassist.Bug448017Test.test_01a(Bug448017Test.java:53)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
```

@szarnekow any idea?

**Comment by szarnekow** [comment]
Apparently a bug in `LightweightTypeReference.isAssignableFrom(JvmType)`

**Comment by szarnekow** [comment]
It should probably read 
```
	public boolean isAssignableFrom(JvmType type) {
		if (type == null) {
			throw new IllegalArgumentException("type may not be null");
		}
		LightweightTypeReference other = getOwner().toLightweightTypeReference(type);
		boolean result = isAssignableFrom(other);
		return result;
	}
```

**Comment by holgerschill** [comment]
This build runs against a fix that I created. Let's see if it is fixed then. https://ci.eclipse.org/xtext/job/xtext-xtend/job/hs_issue677_fix/

**Comment by holgerschill** [comment]
Build is green.

**Comment by cdietrich** [comment]
Fixed in 2.17
